engine/access/rest/http/routes/transactions.go

```diff
@@ -52,6 +52,47 @@ func GetTransactionByID(r *common.Request, backend access.API, link commonmodels
 	return response, nil
 }
 
+// GetTransactionsByBlockID gets transactions by requested blockID.
+func GetTransactionsByBlockID(r *common.Request, backend access.API, link commonmodels.LinkGenerator) (interface{}, error) {
+	req, err := request.NewGetTransactionsByBlockIDRequest(r)
+	if err != nil {
+		return nil, common.NewBadRequestError(err)
+	}
+
+	transactions, err := backend.GetTransactionsByBlockID(r.Context(), req.BlockID)
+	if err != nil {
+		return nil, err
+	}
+
+	var transactionsResponse commonmodels.Transactions
+	// only lookup result if transaction result is to be expanded
+	if req.ExpandsResult {
+		transactionsResponse = make(commonmodels.Transactions, len(transactions))
+
+		for i, transaction := range transactions {
```

Comment:
instead of getting transactions individually, you could use `GetTransactionResultsByBlockID`